server/cluster/cluster.go

```diff
@@ -1046,17 +1046,46 @@ func (c *RaftCluster) GetRangeHoles() [][]string {
 func (c *RaftCluster) UpdateStoreLabels(storeID uint64, labels []*metapb.StoreLabel, force bool) error {
 	store := c.GetStore(storeID)
 	if store == nil {
-		return errors.Errorf("invalid store ID %d, not found", storeID)
+		return errs.ErrInvalidStoreID.FastGenByArgs(storeID)
 	}
 	newStore := proto.Clone(store.GetMeta()).(*metapb.Store)
+	if force {
+		newStore.Labels = labels
+	} else {
+		// If 'force' isn't set, the given labels will merge into those labels which already existed in the store.
+		newStore.Labels = core.MergeLabels(newStore.GetLabels(), labels)
+	}
+	// PutStore will perform label merge.
+	return c.putStoreImpl(newStore)
+}
+
+// DeleteStoreLabel updates a store's location labels
+func (c *RaftCluster) DeleteStoreLabel(storeID uint64, labelKey string) error {
+	store := c.GetStore(storeID)
+	if store == nil {
+		return errs.ErrInvalidStoreID.FastGenByArgs(storeID)
+	}
+	newStore := proto.Clone(store.GetMeta()).(*metapb.Store)
+	labels := make([]*metapb.StoreLabel, 0, len(newStore.GetLabels()))
+	hit := false
+	for _, label := range newStore.GetLabels() {
+		if label.Key == labelKey {
+			hit = true
+			continue
+		}
+		labels = append(labels, label)
+	}
+	if !hit {
```

Comment:
I think we don't need `hit`, can check `if len(labels) == len(newStore.Labels)` instead.